### DISCUSSION TITLE
Not receiving WindowEvent::RedrawRequested after using Window to create Surface with WGPU

### DISCUSSION THREAD

**Discussion by Ominous-Olive** [rust-windowing/winit#3494] [Q&A] [answered]
Hello, I've been playing around with winit and wgpu recently and I encountered an issue causing me to no longer receive `WindowEvent::RedrawRequested` events in the event loop.

Specifically this seems to happen as soon as I pass a reference to the Window to the [`Instance::create_surface`](https://docs.rs/wgpu/latest/wgpu/struct.Instance.html#method.create_surface) method to create a surface.

I'm on MacOS with the current stable toolchain (1.76.0)

The following is a minimal example which causes the issue:

```rust
fn main() {
    let event_loop = winit::event_loop::EventLoop::new().unwrap();
    let window = winit::window::Window::new(&event_loop).unwrap();

    let instance = wgpu::Instance::new(wgpu::InstanceDescriptor {
        ..Default::default()
    });
    // this line causes the issue
    let surface = instance.create_surface(window).unwrap();

    event_loop.run(|event, elwt| match event {
        winit::event::Event::WindowEvent { window_id, event } => match event {
            winit::event::WindowEvent::CloseRequested => elwt.exit(),
            winit::event::WindowEvent::Resized(size) => {
                println!("resize: {:?}", size);
            }
            winit::event::WindowEvent::RedrawRequested => {
                println!("redraw_requested");
            }
            _ => (),
        },
        _ => (),
    });
}
```

**Comment by Ominous-Olive** [comment]
I should clarify, manually calling [Window::request_redraw](https://docs.rs/winit/latest/winit/window/struct.Window.html#method.request_redraw) will still result in a `WindowEvent::RedrawRequested`. What I'm not receiving is redraw requests when resizing the window or similar.

**Comment by madsmtm** [comment, marked as answer]
This was an issue in WGPU, and has been fixed in https://github.com/gfx-rs/wgpu/pull/6107
